[이조은] Add:  Monsters.js, CardList.js, Card.js, SearchBox.js 작업 완료

joyfuljoeunlee opened the pull request:
wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- Task 1~4 완료
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- API 호출
- API 호출의 결과값을 필터링할 수 있는 식과 함께 자식 컴포넌트인 CardList.js에 전달
- map()을 이용하여 배열을 자식 컴포넌트인 Card.js에 전달
- Card.js에서 필요한 데이터 추출

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

jamessoun93 commented on src/2-monsters/index.js:
```diff
@@ -3,41 +3,40 @@ import SearchBox from "./components/SearchBox/SearchBox";
 
-  render() {
-    // 필터링 로직
+    const monstersFilter = this.state.monsters.filter((nombre) => nombre.name.toLowerCase().includes(this.state.userInput));
```
nombre ㅎㅎ 조금 정확히하자면 그 부분에 들어가는 데이터는 monsters 배열에 들어가는 객체 하나하나겠죠? 혹시 헷갈리셨을까봐 커멘트 남겨봅니다!

jamessoun93 replied:
그리고 monster.name 과 userInput 을 둘다 소문자로 변환 후 비교해주셔야 더욱 더 정확한 비교가 됩니다!

junghyunhao replied:
넵! 객체 하나하나에 접근해서 이름을 가져오려고 name.name 하면 헷갈릴까봐 nombre로 했네요 ! 객체 하나에 접근 하는거니.. 다음엔 item 같은 인자를 전달 하면 더 직관적일것 같습니다~ UserInput.toLowerCase로 바꾸도록 하겠습니다~!  감사합니다!^_^

Commit:
Add:  Monsters.js, CardList.js, Card.js, SearchBox.js 작업 완료
- API 호출
- API 호출의 결과값을 필터링할 수 있는 식과 함께 자식 컴포넌트인 CardList.js에 전달
- map()을 이용하여 배열을 자식 컴포넌트인 Card.js에 전달
- Card.js에서 필요한 데이터 추출

jamessoun93 commented on src/Assignments/Monsters/students/11th/jjo-niixx/Components/Card/Card.js:
```diff
@@ -22,11 +22,11 @@ class Card extends Component {
-        <h2>Name</h2>
-        <p>Email</p>
+        <h2>{this.props.filteredCard.name}</h2>
+        <p>{this.props.filteredCard.email}</p>
```
Goood! 👍

jamessoun93 commented on src/Assignments/Monsters/students/11th/jjo-niixx/Components/CardList/CardList.js:
```diff
@@ -11,7 +11,14 @@ import "./CardList.scss";
+        {this.props.monsters.map(filteredList => (
+          <Card 
+            filteredCard = {filteredList}/>
+        ))}
```
`monsters` 배열을 대상으로 map 함수를 호출했을 때, 해당 map 함수의 인자로 들어가는 callback 함수의 인자로 `monster` 배열의 요소가 하나하나 들어온다는 것을 잘 알고 계시지요?!

그렇기 때문에, 이름을 지금처럼 `filteredList` 보다는 `monster` 처럼 한번에 어떤정보인지 파악할 수 있게 지어주시는게 좋습니다.

jamessoun93 commented on src/Assignments/Monsters/students/11th/jjo-niixx/Monsters.js:
```diff
@@ -27,18 +27,30 @@ class Monsters extends Component {
+    fetch("https://jsonplaceholder.typicode.com/users")
+    .then(response => response.json())
+    .then(json => this.setState({monsters: json}))
+    console.log(this.state.monsters)
```
fetch 호출의 response 로 들어오는 결과값을 this.state.monsters 에 저장해주신 후 바로 console.log 로 확인을 하면 업데이트된 값을 받아볼 수 없습니다. 왜 이런 현상이 일어나는지 알고 계실까요? 답변달아주세요~!

joyfuljoeunlee replied:
컴포넌트가 만들어지고 첫 렌더링이 끝나고 componentDidMount 함수가 실행되는데 여기서 state의 값이 바뀌고나서 다시 한번 렌더링이 일어날 때 업데이트된 값을 확인할 수 있습니다. 그러므로 업데이트된 값을 보기 위해서는 render와 return사이에 console.log가 있어야한다고 생각합니다 :)

jamessoun93 commented on src/Assignments/Monsters/students/11th/jjo-niixx/Monsters.js:
```diff
@@ -27,18 +27,30 @@ class Monsters extends Component {
   render() {
+    let filtered = this.state.monsters.filter(data => (
+      data.name.toLowerCase().includes(this.state.userInput)
+    ))
```
data.name 의 소문자 형태와 userInput 을 그대로 비교하는 것 보다 userInput 도 소문자 형태로 변환해준 후 비교해주는 것이 더 정확한 비교대상이 됩니다 :) 

수정해주세요!

jamessoun93 reviewed (changes requested):
리뷰 반영해주세요!

Commit:
Modify: 코드 리뷰 반영
- props 이름 수정
- dataInput과 data.name 비교시 소문자로 통일

jamessoun93 commented:
LGTM!

수고하셨습니다! 🙇🏻‍♂️
